docs/how_to_guides/creating_and_editing_expectations/how_to_create_custom_expectations_for_pandas.rst

```diff
@@ -3,17 +3,256 @@
+
+    In the simplest case, they could be as simple as one-line lambda functions.
+
+    .. code-block:: python
+
+        @MetaPandasDataset.column_map_expectation
+        def expect_column_values_to_be_even(self, column):
+            return column.map(lambda x: x%2==0)
+
+
+    To use the ``column_map_expectation`` decorator, your custom function must accept at least two arguments: ``self`` and ``column``. When the user invokes your Expectation, they will pass a string containing the column name. The decorator will then fetch the appropriate column and pass all of the non-null values to your function as a pandas ``Series``. Your function must then return a Series of boolean values in the same order, with the same index.
+
+    Custom functions can also accept additional arguments:
+
+    .. code-block:: python
+
+        @MetaPandasDataset.column_map_expectation
+        def expect_column_values_to_be_less_than(self, column, value):
+            return column.map(lambda x: x<value)
+
+    Custom functions can have complex internal logic:
+
+    .. code-block:: python
+
+        @MetaPandasDataset.column_map_expectation
+        def expect_column_value_word_counts_to_be_between(self, column, min_value=None, max_value=None):        
+            def count_words(string):
+                word_list = re.findall("(\S+)", string)
+                return len(word_list)
+
+            word_counts = column.map(lambda x: count_words(str(x)))
+
+            if min_value != None and max_value != None:
+                return word_counts.map(lambda x: min_value <= x <= max_value)
+            elif min_value != None and max_value == None:
+                return word_counts.map(lambda x: min_value <= x)
+            elif min_value == None and max_value != None:
+                return word_counts.map(lambda x: x <= max_value)
+            else:
+                return word_counts.map(lambda x: True)
+
+    Custom functions can reference external modules and methods:
+
+    .. code-block:: python
+
+        import pytz
+
+        @MetaPandasDataset.column_map_expectation
+        def expect_column_values_to_be_valid_timezones(self, column, timezone_values=pytz.all_timezones):
+            return column.map(lambda x: x in timezone_values)
+
+    By convention, ``column_map_expectations`` always start with ``expect_column_values_...`` or ``expect_column_value_...`` (Ex: ``expect_column_value_word_counts_to_be_between``). Following this pattern is highly recommended, but not strictly required. If you want to confuse yourself with bad names, the package won't stop you.
+
+
+4. **Load some data**
+
+    To make your new Expectations available for validation, you can instantiate a ``MyCustomPandasDataset`` as follows:
```

Comment:
I think this should be using a Datasource and showing data_asset_type configuration, since that will be the configuration someone has if they go through the tutorial.

Reply:
I considered this. The problem is defining (and iterating on) a `MyCustomPandasDataset` class in a notebook, and then invoking it from within a yaml-configured Datasource.

I don't see a clear path to doing that without making users toggle back and forth between multiple windows and `reload`ing a lot. Any workflow that involves toggling windows is suspect. Any workflow that involves reloading modules in a notebook just plain sucks.

Instead, I believe that users are much more likely to succeed if they do their custom-Expectation-creation dev loop completely within a notebook, then copy the finished class (or methods therein) over to a custom module in `plugins/`. Then they can deploy using yaml-configured Datasources.

This other guide (not yet written) is intended to cover that workflow: https://docs.greatexpectations.io/en/latest/how_to_guides/creating_and_editing_expectations/how_to_invoke_custom_expectations_using_a_datasource.html

(Note: in the future, I can imagine custom tooling to make this workflow slicker.)